packages/node_modules/@node-red/editor-client/src/js/ui/group.js

```diff
@@ -606,7 +606,8 @@ RED.group = (function() {
         for (var i=0; i<nodes.length; i++) {
             n = nodes[i];
             n.dirty = true;
-            var index = group.nodes.indexOf(n);
+            // TODO: Why Proxy indexOf returns -1 ?
```

Comment:
Need to check if `group.nodes` contains raw node objects, or their Proxy objects - and whether `n` is a raw object or Proxy object. If there's any mismatch, then the indexOf won't work because the Proxy object is not the same object as the raw node object.

Ideally everything should be using the Proxy objects under the covers. I'll have a dig.